src/apps/backend/modules/account/internal/account_reader.py

```diff
@@ -70,7 +71,8 @@ def get_account_by_phone_number(*, phone_number: PhoneNumber) -> Account:
 
     @staticmethod
     def check_phone_number_not_exist(*, phone_number: PhoneNumber) -> None:
-        account = AccountRepository.collection().find_one({"phone_number": phone_number, "active": True})
+        phone_number_dict=phone_number.__dict__ if isinstance(phone_number,PhoneNumber) else phone_number
```

Comment:
Why is this check needed when the typdef of argument being passed is `PhoneNumber`- `if isinstance(phone_number,PhoneNumber)`?